addons/naisjobs.md

```diff
@@ -88,65 +116,576 @@ spec:
+
+#### 2.4 (optional) Vault token path as environment variable
+
+`.spec.template.spec.env[]`
+
+```yaml
+spec:
+  template:
+    spec:
+      containers:
+        - name: ${jobname}
+          image: ${image}
+          ...
+          env:
+            - name: VAULT_TOKEN_PATH
+              value: /var/run/secrets/nais.io/vault/vault_token
+```
+
+If your application uses an environment variable to locate the Vault token path.
+
+### 3. Fetching additional secrets from Vault
+
+Repeat the steps below for each secret you want from Vault.
+
+#### 3.1 Specify the paths where your secret should be mounted to
+ 
+Add a new entry to the list in `.spec.initContainers[0].volumeMounts[]` and `.spec.containers[0].volumeMounts[]`:
+
+```yaml
+spec:
+  template:
+    spec:
+      ...
+      containers:
+        - name: ${jobname}
+          image: ${image}
+          ...
+          volumeMounts:
+            - mountPath: ${mount_path}
+              name: vault-secrets
+              subPath: vault/${mount_path}
+      initContainers:
+        - name: vks-init
+          image: ${image}
+          ...
+          volumeMounts:
+            - mountPath: ${mount_path}
+              name: vault-secrets
+              subPath: vault/${mount_path}
+```
+
+where `${mount_path}` is the path to where the secret should be stored and accessed by your application container.
+
+For example, if you want to fetch a `serviceuser`:
+
+```yaml
+spec:
+  template:
+    spec:
+      ...
+      containers:
+        - name: ${jobname}
+          image: ${image}
+          ...
+          volumeMounts:
+            - mountPath: /var/run/secrets/nais.io/${srvUser}
+              name: vault-secrets
+              subPath: vault/var/run/secrets/nais.io/${srvUser}
+      initContainers:
+        - name: vks-init
+          image: ${image}
+          ...
+          volumeMounts:
+            - mountPath: /var/run/secrets/nais.io/${srvUser}
+              name: vault-volume
+              subPath: vault/var/run/secrets/nais.io/${srvUser}
+```
+
+#### 3.2 Specify where the secret is found in Vault
+
+Now that you've specified the mount path for the secret, you'll need to specify where the secret is found in Vault,
+and 
+
+Add a new entry to the list of `.spec.initContainers[0].args`
+
+```yaml
+spec:
+  template:
+    spec:
+      ...
+      initContainers:
+        - name: vks-init
+          image: ${image}
+          ...
+          args:
+            ...
+            - -cn=secret:${kv_path}:dir=${mount_path},fmt=flatten,retries=1
+```
+
+where 
+
+- `${kv_path}` is the location of the Vault secret
+- `${mount_path}` is the `mountPath` that you specified in step 3.1
+
+For example:
+
+```yaml
+spec:
+  template:
+    spec:
+      ...
+      initContainers:
+        - name: vks-init
+          image: ${image}
+          ...
+          args:
+            ...
+            - -cn=secret:/serviceuser/data/prod/srv-my-app:dir=/var/run/secrets/nais.io/srv-my-app,fmt=flatten,retries=1
```

Comment:
skulle dette eksempelet vært mer eksplisitt med at du viste den forrige `-cn` også?

Reply:
Vurderte det først, men tenkte at det kunne være forvirrende i og med at vi viser hvordan man henter ut _en_ secret fra Vault. Mulig folk misforstår og tror at man må ha med begge `-cn` for hver secret?

Reply:
Enig der